Adding documentation for OpenSSL upgrade (#290)
ipdk-io/networking-recipe: docs/guides/security/openssl-guide.md

# OpenSSL Guide

This document provides information about supported OpenSSL versions and
security details that you need to consider.

> OpenSSL is a package dependency as infrap4d uses the library for gRPC.

Reviewer: ```suggestion
OpenSSL is a package dependency, as infrap4d uses the library for gRPC.
```

## End of Life for OpenSSL 1.1.1

OpenSSL 1.1.1 has reached End of Life (EOL) in September 2023. 

It is highly recommended that you upgrade OpenSSL from 1.1.1x to OpenSSL 3.x.

> The official migration guide is [available here](https://www.openssl.org/docs/man3.0/man7/migration_guide.html).

Reviewer: Integral links are better than ones that call attention to themselves (e.g., by saying "here").

```suggestion
See the [official migration guide](https://www.openssl.org/docs/man3.0/man7/migration_guide.html)
for more information.
```

> Starting Fedora 37, Ubuntu 22.04 and Rocky Linux 9.0 OpenSSL 3.0.x comes

Reviewer: ```suggestion
Beginning with Fedora 37, Ubuntu 22.04, and Rocky Linux 9.0, OpenSSL 3.0.x comes
```

standard and requires no further action.

Older distributions of Linux systems download and install OpenSSL 1.1.1 when

> running `yum install` or `apt install` commands. If you are using these

Reviewer: ```suggestion
you run the `yum install` or `apt install` command. If you are using one of these
```

> operating systems, you will need to upgrade the installed package (either find

Reviewer: ```suggestion
distributions, you will need to find an RPM or DEB package to install
```

> an RPM to install upgrade or compile from source).

Reviewer: ```suggestion
or build OpenSSL 3.x from source.
```

> Note that infrap4d will compile and run normally with OpenSSL 1.1.1 since

Reviewer: ```suggestion
Note that infrap4d will compile and run normally with OpenSSL 1.1.1, since
```

> OpenSSL 3.0 is backwards-compatible. In the interest of best security practices

Reviewer: ```suggestion
OpenSSL 3.0 is backward compatible. In the interest of following best security practices
```

> and future security risks, we still recommend upgrading to OpenSSL 3.0.

Reviewer: ```suggestion
and avoiding future security issues, we recommend upgrading to OpenSSL 3.0.
```

# OpenSSL Guide

This document provides information about supported OpenSSL versions and
security details that you need to consider.

OpenSSL is a package dependency, as infrap4d uses the library for gRPC.

## End of Life for OpenSSL 1.1.1

OpenSSL 1.1.1 has reached End of Life (EOL) in September 2023. 

> It is highly recommended that you upgrade OpenSSL from 1.1.1x to OpenSSL 3.x.

Reviewer: ```suggestion
It is highly recommended that you upgrade OpenSSL from 1.1.x to OpenSSL 3.x.
```

See the [official migration guide](https://www.openssl.org/docs/man3.0/man7/migration_guide.html)
for more information.

Beginning with Fedora 37, Ubuntu 22.04, and Rocky Linux 9.0, OpenSSL 3.0.x comes
standard and requires no further action.

Older distributions of Linux systems download and install OpenSSL 1.1.1 when
you run the `yum install` or `apt install` command. If you are using one of these

> distributions, you will need to find an RPM or DEB package to install

Reviewer: This is difficult to parse. I would suggest one of the following:

"you will need to (a) find an RPM or DEB package to install, or (b) download OpenSSL 3.x and build it from source."

"you will need to find an RPM or DEB package to install, or else download OpenSSL 3.x and build it from source."
